tests/test_notify_suppliers_of_brief_withdrawal.py

```diff
@@ -0,0 +1,94 @@
+
+
+@mock.patch('dmutils.email.DMNotifyClient', autospec=True)
+@mock.patch('dmscripts.notify_suppliers_of_brief_withdrawal.create_context_for_brief')
+@mock.patch('dmscripts.notify_suppliers_of_brief_withdrawal.get_brief_response_emails')
+@mock.patch('dmapiclient.DataAPIClient', autospec=True)
+def test_main(data_api_client, get_brief_response_emails, create_context_for_brief, notify_client):
+    data_api_client.find_briefs.return_value = WITHDRAWN_BRIEFS
+    tested_script.create_context_for_brief.return_value = EXPECTED_BRIEF_CONTEXT
+    get_brief_response_emails.side_effect = [["email@me.now", "email@them.now"], []]
+    withdrawn_date = date(2016, 1, 28)
+    tested_script.main(data_api_client, notify_client, "notify_template_id", "preview", mock.Mock(), withdrawn_date)
+    assert data_api_client.find_briefs.call_args_list == [mock.call(withdrawn_on=withdrawn_date)]
+    assert create_context_for_brief.call_args_list == [mock.call("preview", WITHDRAWN_BRIEFS["briefs"][0])]
+    assert notify_client.send_email.call_args_list == [
+        mock.call("email@me.now", "notify_template_id", EXPECTED_BRIEF_CONTEXT, allow_resend=False),
+        mock.call("email@them.now", "notify_template_id", EXPECTED_BRIEF_CONTEXT, allow_resend=False)
+    ]
+
+
+@mock.patch('dmutils.email.DMNotifyClient', autospec=True)
+@mock.patch('dmapiclient.DataAPIClient', autospec=True)
+def test_main_calls_correct_client_methods(data_api_client, notify_client):
+
+    data_api_client.find_briefs.return_value = WITHDRAWN_BRIEFS
+    data_api_client.find_brief_responses.side_effect = (BRIEF_RESPONSES, {"briefResponses": []})
+    withdrawn_date = date.today()
+
+    result = tested_script.main(
+        data_api_client, notify_client, 'notify_template_id', 'preview', mock.Mock(), withdrawn_date
+    )
+
+    assert result is True
+    assert data_api_client.find_briefs.call_args_list == [mock.call(withdrawn_on=withdrawn_date)]
+    assert notify_client.send_email.call_args_list == [
+        mock.call("email@me.now", "notify_template_id", EXPECTED_BRIEF_CONTEXT, allow_resend=False),
+        mock.call("email@them.now", "notify_template_id", EXPECTED_BRIEF_CONTEXT, allow_resend=False)
+    ]
+
+
+@mock.patch('dmutils.email.DMNotifyClient', autospec=True)
+@mock.patch('dmapiclient.DataAPIClient', autospec=True)
+def test_single_call_when_brief_id_specified(data_api_client, notify_client):
+    """Script should only look up brief responses for given brief id when brief id is specified."""
+    data_api_client.find_briefs.return_value = WITHDRAWN_BRIEFS
+    brief_id = 235
+    result = tested_script.main(
+        data_api_client, notify_client, 'notify_template_id', 'preview', mock.Mock(), date.today(), brief_id=brief_id
+    )
+
+    assert result is True
+    data_api_client.find_brief_responses.assert_called_once_with(brief_id=brief_id, status='submitted')
```

Comment:
Should we also assert that we only send one email?

Reply:
So the idea is that the part of the script that I am testing here is the part that gets the brief responses and therefore the emails. 
At this point it could send any number of emails and still be valid but it should only make one call to the brief responses end point.

Reply:
Ahh you're right of course - multiple suppliers could apply to one brief, that makes sense now. I think I got confused because we don't assert the call for `data_api_client.get_brief_response_emails` when it's called multiple times (because we're mocking out the helper function?).